tests/test_util.py

```diff
@@ -20,38 +20,184 @@
+    assert IntervalSchedule(interval=1.0, initial_delay=0.5) != IntervalSchedule(interval=1.0, initial_delay=0.0)
+
+
+def test_cron_schedule(monkeypatch) -> None:
+    def freeze_time(time: datetime.datetime) -> None:
+        """
+        Freeze time to a given value to avoid race conditions.
+        """
+        datetime_orig = datetime.datetime
+
+        class FrozenDatetimeMeta(type(datetime_orig)):
+            # mock isinstance checks on datetime objects
+            def __instancecheck__(self, instance):
+                return super().__instancecheck__(instance) or isinstance(instance, datetime_orig)
+
+        class frozendatetime(datetime.datetime, metaclass=FrozenDatetimeMeta):
+            @classmethod
+            def now(cls, tz: Optional[datetime.tzinfo] = None) -> datetime.datetime:
+                if tz is None:
+                    return time if time.tzinfo is None else time.astimezone(datetime.timezone.utc).replace(tzinfo=None)
+                else:
+                    aware: datetime.datetime = time if time.tzinfo is not None else time.replace(tzinfo=datetime.timezone.utc)
+                    return aware.astimezone(tz)
+
+        monkeypatch.setattr(datetime, "datetime", frozendatetime)
+
+    # min hour day month dow
+    schedule: TaskSchedule = CronSchedule(cron="1 2 * * *")
+
+    # year month day hour min sec ms
+    now: datetime.datetime = datetime.datetime(2022, 6, 16, 10, 15, 3, 0)
+    freeze_time(now)
+    next_delta: datetime.timedelta = datetime.datetime(2022, 6, 17, 2, 1, 0, 0) - now
+    assert schedule.get_initial_delay() == schedule.get_next_delay() == next_delta.total_seconds()
+
+    tomorrow: datetime.datetime = now + datetime.timedelta(days=1, seconds=42)
+    freeze_time(tomorrow)
+    assert schedule.get_initial_delay() == schedule.get_next_delay() == next_delta.total_seconds() - 42
+
+
+def test_cron_schedule_equals() -> None:
+    assert CronSchedule(cron="1 2 * * *") == CronSchedule(cron="1 2 * * *")
+    assert CronSchedule(cron="1 2 * * *") != CronSchedule(cron="0 2 * * *")
+
+
 async def test_scheduler_remove(caplog):
-    sched = util.Scheduler("remove")
+    scheduler = util.Scheduler("remove")
 
     i = []
 
     async def action():
         i.append(0)
 
-    sched.add_action(action, 0.05, 0)
+    schedule: IntervalSchedule = IntervalSchedule(0.05, 0)
+    scheduler.add_action(action, schedule)
 
     while len(i) == 0:
         await asyncio.sleep(0.01)
 
-    sched.remove(action)
+    assert (action, schedule) in scheduler._scheduled
+    scheduler.remove(action, schedule)
     length = len(i)
     await asyncio.sleep(0.1)
     assert len(i) == length
-    assert not sched._executing_tasks[action]
+    assert not scheduler._executing_tasks[action]
+    assert not (action, schedule) in scheduler._scheduled
     no_error_in_logs(caplog)
 
 
+async def test_scheduler_remove_same_action() -> None:
```

Comment:
This test case can use a docstring.

Reply:
I thought I'd fully gotten into the habit of adding docstrings to all testcases. Will fix.